.github/workflows/codeql-query.yml

```diff
@@ -0,0 +1,67 @@
+name: CodeQL query
+
+on:
+  workflow_call:
+    secrets:
+      ACTIONS_STEP_DEBUG:
+        description: step debug
+
+jobs:
+  run-query:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        repoIds: ${{ fromJSON(github.event.inputs.repoIdChunks) }}
+
+    steps:
+      # Use ::add::mask for all tokens.
+      # Theoretically we'll only output a subset of the tokens. Unless it causes a
+      # performance issue, we may as well mask all tokens to be extra safe.
+      # Note that masking does not persist to other/future jobs in the same workflow.
+      - name: Mask tokens
+        run: |
+          jq .inputs.repositories "$GITHUB_EVENT_PATH" --raw-output | \
+            jq '[.[].token , .[].pat] | map( select( . != null ) ) | .[]' --raw-output | \
```

Comment:
I updated this to add pat masking as well because I was testing from the command line. However, it seems like Actions may recognise the PAT format and mask them anyway because I made a mistake with my inputs and they were still filtered out: https://github.com/dsp-testing/qc-run2/runs/3755154364?check_suite_focus=true

Note: the two jq calls should be able to be folded into one using a jq pipe filter.

Reply:
Yes, actions masks PATs or other things it recognises as a github token automatically. Still though, the explicit masking looks good to me.